### PR TITLE
Feature mass cancel running actions on ds invalidation

### PR DESCRIPTION
**Pull request by Nkyn**
A feature to invalidate distribution sets was introduced recently. 
This is an extension to that feature, that will give the user the option to also stop all running actions affected by the invalidation. An option is added to the invalidation dialogue in the UI, whether the action shall be soft cancelled or forced cancelled.

### CODE REVIEW

File: hawkbit-ui/src/main/resources/messages.properties [hunk 1]
```diff
@@ -369,6 +379,12 @@ tooltip.target.status.pending = Status pending
 tooltip.target.status.registered = Status registered
 tooltip.target.pin = Pin target
 tooltip.distribution.set.pin = Pin distribution set
+tooltip.invalidate.distributionset = Invalidate distribution set
+tooltip.distributionset.invalid = invalid
+tooltip.distributionset.incomplete = incomplete
+tooltip.distributionset.invalidate.forced = Cancelation of open actions will be forced
```
Comment (sebastian-firsching) [inline hawkbit-ui/src/main/resources/messages.properties:385]: Maybe phrase it like "Open actions will be canceled and force quit" as we also use "force quit" in the action history grid

File: hawkbit-ui/src/main/resources/messages.properties [hunk 2]
```diff
@@ -501,6 +518,17 @@ message.forcedTime.cannotBeEmpty = Forced time can not be empty
...
+message.invalidate.distributionset.consequences.singular = By invalidating this distribution set it can no longer be used in any assignment. All active auto assignments and optionally all active rollouts of the distribution set can be stopped as well as all existing update actions will be canceled. A summary of affected entities will be shown on the next page.
+message.invalidate.distributionset.consequences.plural = By invalidating this {0} distribution sets they can no longer be used in any assignment. All active auto assignments and optionally all active rollouts of the distribution sets can be stopped as well as all existing update actions will be canceled. A summary of affected entities will be shown on the next page.
+message.invalidate.distributionset.affected.entities.intro.singular = When you confirm the invalidation of distribution set {0} the following actions will be taken:
+message.invalidate.distributionset.affected.entities.intro.plural = When you confirm the invalidation of {0} distribution sets the following actions will be taken:
+message.invalidate.distributionset.affected.entities.actions = {0} action(s) will be stopped
```
Comment (Nkyn) [inline hawkbit-ui/src/main/resources/messages.properties:529]:
```suggestion
message.invalidate.distributionset.affected.entities.actions = {0} action(s) will be cancelled
```

File: hawkbit-ui/src/main/resources/messages.properties [hunk 3]
```diff
@@ -501,6 +518,17 @@ message.forcedTime.cannotBeEmpty = Forced time can not be empty
...
+message.invalidate.distributionset.affected.entities.rollouts = {0} rollout(s) will be stopped
+message.invalidate.distributionset.affected.entities.autoassignments = {0} auto assignment(s) will be stopped
```
Comment (Nkyn) [inline hawkbit-ui/src/main/resources/messages.properties:531]:
```suggestion
message.invalidate.distributionset.affected.entities.autoassignments = {0} auto assignment(s) will be removed
```

File: hawkbit-ui/src/main/resources/messages.properties (hunk 1 above)
Comment (Nkyn) [reply hawkbit-ui/src/main/resources/messages.properties]: Done - changed the tooltip string as suggested

File: hawkbit-ui/src/main/resources/messages.properties (hunk 2 above)
Comment (Nkyn) [reply hawkbit-ui/src/main/resources/messages.properties]: Done

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/dstable/InvalidateDistributionSetSupport.java [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+
+        long affectedActionsByDSInvalidation = 0;
+        if (consequencesDialog.getCancelationType() == CancelationType.FORCE) {
+            affectedActionsByDSInvalidation = getAffectedActionsByForcedDSInvalidation(
+                    allDistributionSetsForInvalidation);
+        } else if (consequencesDialog.getCancelationType() == CancelationType.SOFT) {
+            affectedActionsByDSInvalidation = getAffectedActionsBySoftDSInvalidation(
+                    allDistributionSetsForInvalidation);
+        }
+
```
Comment (Nkyn) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/dstable/InvalidateDistributionSetSupport.java:100]: if else should be part of a lower lever to ease the call of it for the client.

File: hawkbit-ui/src/main/resources/messages.properties (hunk 3 above)
Comment (Nkyn) [reply hawkbit-ui/src/main/resources/messages.properties:531]: Won't fix.

**Review by sebastian-firsching — APPROVED** [review]

**Review by laverman — APPROVED** [review]

**Pull request merged**